STYLE_GUIDE.md

```diff
@@ -0,0 +1,1245 @@
+
+### DON'T have HRX files longer than 500 lines or so
+
+Very large HRX files can be as difficult to naviate as hundreds of tiny files.
+Once an HRX file gets larger than about 500 lines, it's probably time to split
+it into multiple files. To do so:
+
+* Create a directory with the same name as the HRX files.
+* Take all the top-level subdirectories in the HRX file and make them their own
+  HRX files.
+* If this creates too many small HRX files, consider combining some related
+  files back into more coarse-grained sub-divisions.
+
+### DO use descriptive paths
+
+<details>
+<summary>Example</summary>
+
+#### Good
+
+`css/keyframes.hrx`
+
+```hrx
+<===> bubble/empty/input.scss
+// Regression test for sass/dart-sass#611.
+a {
+  @keyframes {/**/}
+}
+
+<===> bubble/empty/output.css
+@keyframes {
+  /**/
+}
+```
+
+#### Bad
+
+`regression/dart_sass_611.hrx`
+
+```hrx
+<===> input.scss
+a {
+  @keyframes {/**/}
+}
+
+<===> output.css
+@keyframes {
+  /**/
+}
+```
+
+</details>
+
+The path to a spec, both inside and outside its HRX file, is the first place a
+reader will look to try to understand [exactly what it's testing][]. Being as
+descriptive as possible (without being too verbose) in your choice of names will
+make this understanding much easier. Good rules of thumb include:
+
+[exactly what it's testing]: #do-test-only-one-thing-per-spec
+
+* Describe the feature or behavior being tested, not the circumstances around
+  how the test came to be created.
+
+* Use noun phrases (like "keyframes") or adjectives (like "empty") that describe
+  the feature being tested or the context or state it's in.
+
+* Avoid placeholder names like "basic" or "simple". Sometimes it's best to
+  describe *how* it's basic, like "empty" or "one_child". Other times, you can
+  just forego the "basic" directory entirely and put your specs beneath the
+  feature itself.
+
+If you can't fully and tersely describe a spec with its name alone, [add a
+comment][] to the beginning of the Sass file.
+
+[add a comment]: #consider-adding-a-comment-explaining-your-spec
+
+### DO put error specs in a separate "error" directory
+
+<details>
+<summary>Example</summary>
+
+#### Good
+
+```hrx
+<===> single_value/input.scss
+a {
+  --b: c;
+}
+
+<===> single_value/output.css
+a {
+  --b: c;
+}
+
+<===>
+================================================================================
+<===> error/empty/input.scss
+// CSS requires at least one token in a custom property.
+.empty {
+  --property:;
+}
+
+<===> error/empty/error
+Error: Expected token.
+  ,
+3 |   --property:;
+  |              ^
+  '
+  /sass/spec/css/custom_properties/error/empty/input.scss 3:14  root stylesheet
+```
+
+#### Bad
+
+```hrx
+<===> single_value/input.scss
+a {
+  --b: c;
+}
+
+<===> single_value/output.css
+a {
+  --b: c;
+}
+
+<===>
+================================================================================
+<===> empty/input.scss
```

Comment:
Should this be `error/empty/input.scss`?

Reply:
Yes; this is the "bad" example, intended to indicate the incorrect way to do things.